deepchem/molnet/load_function/molnet_loader.py

```diff
@@ -0,0 +1,207 @@
+"""
+Common code for loading MoleculeNet datasets.
+"""
+import os
+import logging
+import deepchem as dc
+from deepchem.data import Dataset, DiskDataset
+from typing import List, Optional, Tuple, Type, Union
+
+logger = logging.getLogger(__name__)
+
+
+class TransformerGenerator(object):
+  """Create Transformers for Datasets.
+
+  When loading molnet datasets, you cannot directly pass in Transformers
+  to use because many Transformers require the Dataset they will be applied to
+  as a constructor argument.  Instead you pass in TransformerGenerator objects
+  which can create the Transformers once the Dataset is loaded.
+  """
+
+  def __init__(self, transformer_class: Type[dc.trans.Transformer], **kwargs):
+    """Construct an object for creating Transformers.
+
+    Parameters
+    ----------
+    transformer_class: Type[Transformer]
+      the class of Transformer to create
+    kwargs:
+      any additional arguments are passed to the Transformer's constructor
+    """
+    self.transformer_class = transformer_class
+    self.kwargs = kwargs
+
+  def create_transformer(self, dataset: Dataset) -> dc.trans.Transformer:
+    """Construct a Transformer for a Dataset."""
+    return self.transformer_class(dataset=dataset, **self.kwargs)
+
+  def get_directory_name(self) -> str:
+    """Get a name for directories on disk describing this Transformer."""
+    name = self.transformer_class.__name__
```

Comment:
Should we add a string method for transformers the same way we have for splitters/featurizers? That might allow for more fine grained naming control in case there are transformer options that lead to different transformations

Reply:
This is essentially the same string representation we use for splitters and featurizers: `classname_arg_value_arg_value`.  The code is a little different because we're describing a transformer that hasn't been created yet, so it can't go in the transformer's `__str__()` method.